subprojects/core/src/test/groovy/org/gradle/internal/fingerprint/impl/EmptyFileCollectionFingerprintTest.groovy

```diff
@@ -45,20 +55,26 @@ class EmptyFileCollectionFingerprintTest extends Specification {
             "file2.txt": new DefaultNormalizedFileSnapshot("file2.txt", FileType.RegularFile, HashCode.fromInt(234)),
         ], FingerprintCompareStrategy.ABSOLUTE)
         expect:
-        getChanges(EmptyFileCollectionFingerprint.INSTANCE, fingerprint, false).toList() == [
+        getChanges(empty, fingerprint, false).toList() == [
             FileChange.removed("file1.txt", "test", FileType.RegularFile),
             FileChange.removed("file2.txt", "test", FileType.RegularFile)
         ]
-        getChanges(EmptyFileCollectionFingerprint.INSTANCE, fingerprint, true).toList() == [
+        getChanges(empty, fingerprint, true).toList() == [
             FileChange.removed("file1.txt", "test", FileType.RegularFile),
             FileChange.removed("file2.txt", "test", FileType.RegularFile)
         ]
+
+        where:
+        empty << EMPTY_FINGERPRINTS
     }
 
     def "comparing to itself works"() {
         expect:
-        getChanges(EmptyFileCollectionFingerprint.INSTANCE, EmptyFileCollectionFingerprint.INSTANCE, false).toList() == []
-        getChanges(EmptyFileCollectionFingerprint.INSTANCE, EmptyFileCollectionFingerprint.INSTANCE, true).toList() == []
+        getChanges(empty, empty, false).toList() == []
+        getChanges(empty, empty, true).toList() == []
+
+        where:
+        empty << EMPTY_FINGERPRINTS
```

Comment:
I would prefer comparing combinations of empty fingerprints, not only fingerprints to themselves. Normally, a `CurrentEmptyFileCollectionFingerprint` would be compared to a `HistoricalEmptyFileCollectionFingerprint`.